tools/protoxform/protoprint.py

```diff
@@ -162,13 +167,16 @@ def format_type_context_comments(type_context, annotation_xforms=None):
     return leading, trailing
 
 
-def format_header_from_file(source_code_info, file_proto, empty_file):
+def format_header_from_file(source_code_info, file_proto, empty_file, import_deprecation_proto,
+                            frozen_proto):
     """Format proto header.
 
     Args:
         source_code_info: SourceCodeInfo object.
         file_proto: FileDescriptorProto for file.
         empty_file: are there no message/enum/service defs in file?
+        import_deprecation_proto: should the "envoy/annotation/deprecation.proto" be imported?
+        frozen_proto: is the proto file frozen?
```

Comment:
The logic here is a bit hard to follow; could we have something more descriptive, e.g. `has_minor_version_deprecation` and then infer these other properties?

Reply:
Done.